### PR TITLE
Allow backlight to use multiple backlight commands, add configuration for light

### PR DESCRIPTION
**Pull request by anubiann00b**

### CODE REVIEW

File: py3status/modules/backlight.py [hunk 1]
```diff
@@ -3,6 +3,10 @@
 Adjust screen backlight brightness.
 
 Configuration parameters:
+    backlight_command: The program to use to change the backlight.
```
Comment (lasers) [inline py3status/modules/backlight.py]: Just `command` here would be nice.

File: py3status/modules/backlight.py [hunk 2]
```diff
@@ -81,18 +98,22 @@ def post_config_hook(self):
...
+            except ValueError:
+                self.backlight_command_available = False
         except:
             pass
 
-        if self.xbacklight and self.brightness_initial:
+        if self.backlight_command_available and self.brightness_initial:
             self._set_backlight_level(self.brightness_initial)
 
     def on_click(self, event):
```
Comment (lasers) [inline py3status/modules/backlight.py]: I'm only using `light` right now... Scrolling here does not change values for me?

File: py3status/modules/backlight.py (hunk 1 above)
Comment (anubiann00b) [reply py3status/modules/backlight.py]: Fixed

File: py3status/modules/backlight.py (hunk 2 above)
Comment (anubiann00b) [reply py3status/modules/backlight.py]: It works for me now... Perhaps this was the issue? https://github.com/ultrabug/py3status/pull/1077/commits/5cee59a27b38f311743e4c8cc51518d0516effc1#diff-7d7c9df691aab6e0ddfbacd9e0c81b94R133

I'm not super familiar with python

**Comment by anubiann00b** [comment]
I can squash this if it's good to go

**Comment by lasers** [comment]
No rush. I think others will want to review this too. I struggle to scroll down/up with `light` with 7 raw values. I think it's because of `light -G` that only can return 7 different values... so when I scroll, I kept getting the same value in the beginning of `on_click`.

I guess I can scroll down, then I'm stuck there... Can't scroll up at all. You might be able to notice this if you play with `light -G` and `light -S 44` (44, 45, 46, 47, 48, etc) where `light -G` does not change the percent number. `light` ignore percent if it can't exceed raw numbers %. Anyhow, that's what I think the issue was. With `xbacklight`, everything seems to be fine... Working great.

**Comment by anubiann00b** [comment]
What's this with light -G returning 7 values...? It works on my machine the way it's supposed to, going off of percent. Maybe it's taking the numbers as raw values somewhere on your machine? I think there's a flag to specify percent or raw values so that probably should be added.

**Comment by lasers** [comment]
Run `light -Grm`. You probably have over `1000`. This laptop have `7`. 
```
$ light -Grm
7
```
```
$ light -G && light -S 50
42.86
$ light -G && light -S 51
42.86
$ light -G && light -S 52
42.86
$ light -G && light -S 53
42.86
$ light -G && light -S 54
42.86
$ light -G && light -S 55
42.86
$ light -G && light -S 56
42.86
$ light -G && light -S 57
42.86
$ light -G && light -S 58
42.86
```
```
$ light -G && light -S 59
57.14
$ light -G && light -S 65
57.14
```
```
100/7 = 14.28~ each
14.28~
28.57~
42.85~
57.14~
71.42~
85.71~
100.0~
```
>Maybe it's taking the numbers as raw values somewhere on your machine?

From `light -Grm`... or 
```
cat /sys/class/backlight/acpi_video0/max_brightness
cat /sys/class/backlight/acpi_video0/brightness
```
> I think there's a flag to specify percent or raw values so that probably should be added.

`light -A/-U` + `delta`. I think `xbacklight`'s delta works with percent and for light, its delta could be working okay w/ your laptop (is `percent`) or my old laptop (is `raw`). 😕

**Comment by anubiann00b** [comment]
Oh I see... I'm a bit confused by your proposed solution though? Are you suggesting adding the raw value via `light -Ar [delta]`?

**Comment by lasers** [comment]
I'm not sure...

I think we might be better off sticking with percent because everybody's values are different.

Proposed solution could be... Make `self.cache_percent` so we can track the `percent` number in the `on_click` rather than getting it from `level = self._get_backlight_level()` because `light` would give back wrong / same percent number... Doing this way should give us a decent `percent` number to work with? Something might be a little off when we start adjusting. Maybe not a big deal. Could try using `delta` 5 too... so we scroll up/down a bit more nice number.

**Comment by tobes** [comment]
@anubiann00b 

I like the general look of this PR but I'd make the following changes.

I'd default the `command` param to `None` and in that case try to see which command we can use.  If it is set to `xbacklight` we should use that method even if broken imho.  This way it should work for users with them doing nothing.

> Make `self.cache_percent`

This works until the user changes the brightness via another method.  From what I can tell reading here we have enough info eg from `light -Grm` etc to know what level we need to actually change the brightness.  We should respect `brightness_delta` param where we can, but if it is too small then we should just increase the brightness by the minimal amount.

**Comment by maximbaz** [comment]
This patch works really well for me, however my value of `light -Grm` is `1500` 🙂

**Comment by maximbaz** [comment]
Hey @anubiann00b, would you have time to finalize this PR? I'm using your patch locally for a month or so, works perfectly, but I'd love to see it finally merged 🙂

**Comment by tobes** [comment]
@anubiann00b Hi I'd also be keen to get this merged

**Comment by anubiann00b** [comment]
Re: the last 4 comments and https://github.com/ultrabug/py3status/pull/1071#issuecomment-359194441

I'm not really able to work on this right now, sorry. Forking this and adding the final fixes is probably the best way to go.

**Comment by maximbaz** [comment]
@lasers since it's only you who could reproduce issue with low `Grm` value, maybe you could take over this PR and get it merged?

**Comment by tobes** [comment]
>  get it merged?

It'd be nice to close this yes :)

**Comment by ultrabug** [comment]
Any update here please?

**Comment by lasers** [comment]
This requires parties with working `xbacklight` and working `light`. I'm thinking about ignoring the low `light` values because that could be just an old laptop. Newer sensible laptops probably don't have this issue... but it's good to know that we "sorta" can't exactly trust `light` with scrolling.

**Comment by maximbaz** [comment]
How about then merging this now and tackling future improvements in a separate PR?

**Comment by lasers** [comment]
I can tackle this after we're done simplifying and cleaning up things in `sysdata`.

**Comment by ultrabug** [comment]
Thanks everyone, let's continue from there

**Comment by maximbaz** [comment]
I'm confused, this is shown as merged, but the change is not actually in [master](https://github.com/ultrabug/py3status/blob/master/py3status/modules/backlight.py)? 
And sadly not in 3.8 either 😞

**Pull request merged**